ISM43362Interface.cpp

```diff
@@ -446,7 +446,7 @@ int ISM43362Interface::socket_recv(void *handle, void *data, unsigned size)
             socket->connected = false;
             debug_if(_ism_debug, "ISM43362Interface socket_recv: socket closed\r\n");
             _mutex.unlock();
-            return NSAPI_ERROR_CONNECTION_LOST;
+            return 0;
```

Comment:
but here, this sounds more like a network loss. 
Have you checked in #29 if only the previous return is enough to solve the issue ?